File: backend/app/schemas/chord_simplification.py
from typing import Literal

from pydantic import BaseModel


DifficultyId = Literal[
    "absolute-beginner",
    "beginner",
    "developing",
]


ChordDecisionAction = Literal[
    "practice",
    "defer",
    "review",
]


class ChordDecision(BaseModel):
    chord: str

    action: ChordDecisionAction

    difficulty_score: int | None = None

    reason: str


class ChordSimplificationResult(BaseModel):
    difficulty: DifficultyId

    original_chords: list[str]

    practice_chords: list[str]

    deferred_chords: list[str]

    review_chords: list[str]

    decisions: list[ChordDecision]

    recommended_bpm: int

    max_chords_per_session: int

    manual_review_required: bool

    notes: list[str]

File: src/types/chordSimplification.ts
export type ChordDecisionAction = "practice" | "defer" | "review";

export type ChordDecision = {
  chord: string;

  action: ChordDecisionAction;

  difficulty_score: number | null;

  reason: string;
};

export type ChordSimplification = {
  difficulty: "absolute-beginner" | "beginner" | "developing";

  original_chords: string[];

  practice_chords: string[];

  deferred_chords: string[];

  review_chords: string[];

  decisions: ChordDecision[];

  recommended_bpm: number;

  max_chords_per_session: number;

  manual_review_required: boolean;

  notes: string[];
};
